### PR TITLE
Release 12.0.0a1

### LINKED ISSUE #1052: Cut version 12.0.0 alpha for client
Kubernetes Python client v12.0.0 would track Kubernetes v1.16

### Tasks

- [x] Cut 12.0.0 snapshot branch
- [x] Upgrade openapi-generator version
- [x] Drop Python 2 support
> To be updated

### PR DESCRIPTION
**Pull request by scottilee**
Release 12.0.0a1 introduces several changes and is compatible with Kubernetes v1.16.

[x] Updated the constants version
[x] Generated client changes
[x] Applied hotfixes
[x] Updated python-base submodules
[x] Updated the CHANGELOG

Fixes #1052

### CODE REVIEW

File: kubernetes/client/api/apps_v1_api.py [hunk 1]
```diff
@@ -3887,7 +3887,7 @@ def patch_namespaced_deployment_with_http_info(self, name, namespace, body, **kw
 
         # HTTP header `Content-Type`
         header_params['Content-Type'] = self.api_client.select_header_content_type(  # noqa: E501
-            ['application/json-patch+json', 'application/merge-patch+json', 'application/strategic-merge-patch+json'])  # noqa: E501
+            ['application/json-patch+json', 'application/merge-patch+json', 'application/strategic-merge-patch+json', 'application/apply-patch+yaml'])  # noqa: E501
```
Comment (nxexox) [inline kubernetes/client/api/apps_v1_api.py:3890]: Hi. I have `K8S 1.16`. I noticed one interesting bug. If I call `patch_namespaced_deployment` through the client to `DELETE` 1 or more environment variables, then the request is successful, however, the deployment itself is not updated.

After a little research, I found out that the `select_header_content_type` method called here returns just the 1st element of this array. I also found out that if you put the 2nd element of an array in the header, then the deployment is updated perfectly.

```suggestion
            ['application/merge-patch+json', 'application/json-patch+json', 'application/strategic-merge-patch+json', 'application/apply-patch+yaml'])  # noqa: E501
```

File: scripts/constants.py [hunk 2]
```diff
@@ -15,10 +15,10 @@
 import sys
 
 # Kubernetes branch to get the OpenAPI spec from.
-KUBERNETES_BRANCH = "release-1.15"
+KUBERNETES_BRANCH = "release-1.16"
 
 # client version for packaging and releasing.
-CLIENT_VERSION = "11.0.0-snapshot"
+CLIENT_VERSION = "12.0.0a1"
```
Comment (roycaihw) [inline scripts/constants.py:21]: This PR targets the master branch. I think we should first generates a `12.0.0-snapshot` client in the master branch with API changes from upstream `release-1.16`. Then generates a `12.0.0a1` client in the `release-12.0` branch with only version constant changes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
+# v12.0.0a1
+
+ Kubernetes API Version: 1.16
+
+ **Deprecation Notice**
+v12.0.0 of the client follows the Kubernetes [deprecation policy](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#supported-releases-and-component-skew) and will
+be deprecated as and when Kubernetes version v1.16 gets deprecated.
+
```
Comment (roycaihw) [inline CHANGELOG.md:8]: Could you add CHANGELOG about the API changes generated?

**Comment by scottilee** [comment]
Will open a new PR when https://github.com/kubernetes-client/python/pull/1141 merges.

**Pull request closed without merging**